agent/command/papertrail_trace.go

```diff
@@ -0,0 +1,220 @@
+package command
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	agentinternal "github.com/evergreen-ci/evergreen/agent/internal"
+	agentinternalclient "github.com/evergreen-ci/evergreen/agent/internal/client"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	productionPapertrailAddress = "https://papertrail.devprod-infra.prod.corp.mongodb.com"
+)
+
+var (
+	errMustSpecifyKeyID              = fmt.Errorf("must specify key_id")
+	errMustSpecifySecretKey          = fmt.Errorf("must specify secret_key")
+	errMustSpecifyProduct            = fmt.Errorf("must specify product")
+	errMustSpecifyVersion            = fmt.Errorf("must specify version")
+	errMustSpecifyAtLeastOneFilename = fmt.Errorf("must specify at least one filename")
+)
+
+type papertrailTrace struct {
+	Address   string   `mapstructure:"address"`
+	KeyID     string   `mapstructure:"key_id"`
+	SecretKey string   `mapstructure:"secret_key"`
+	Product   string   `mapstructure:"product"`
+	Version   string   `mapstructure:"version"`
+	Filenames []string `mapstructure:"filenames"`
+	WorkDir   string   `mapstructure:"work_dir"`
+	base
+}
+
+func (t *papertrailTrace) Execute(ctx context.Context,
+	comm agentinternalclient.Communicator, logger agentinternalclient.LoggerProducer, conf *agentinternal.TaskConfig) error {
+	client := &papertrailClient{
+		httpc:     &http.Client{},
+		address:   t.Address,
+		keyID:     t.KeyID,
+		secretKey: t.SecretKey,
+	}
+
+	// TODO: change to submitter when available (DEVPROD-5223)
+	author, ok := conf.Expansions["author"]
+	if !ok {
+		return fmt.Errorf("could not find author in expansions")
+	}
+
+	taskID, ok := conf.Expansions["task_id"]
+	if !ok {
+		return fmt.Errorf("could not find task_id in expansions")
+	}
+
+	execution, ok := conf.Expansions["execution"]
+	if !ok {
+		return fmt.Errorf("could not find task_id in expansions")
+	}
+
+	papertrailBuildID := fmt.Sprintf("%s_%s", taskID, execution)
+
+	workdir := t.WorkDir
+	if workdir == "" {
+		workdir = conf.WorkDir
+	}
+
+	for _, file := range t.Filenames {
+		fullname := filepath.Join(workdir, file)
```

Comment:
I know this isn't ready for review yet, but just because I happened to notice it and don't want to forget: 
You need something like `fullname, err := conf.Expansions.ExpandString(file)` before this because the user might use an expansion when specifying the filename in their config file. We also have a GetWorkingDirectory function that you can use.